telegram/utils/request.py

```diff
@@ -64,15 +65,16 @@ class Request(object):
             consecutive read operations for a response from the server. None will set an infinite
             timeout. This value is usually overridden by the various ``telegram.Bot`` methods.
             (default: 5.)
-
+        retries (int): Number of retries for each request. Use 0 to disable retrying. (default: 3)
     """
 
     def __init__(self,
                  con_pool_size=1,
                  proxy_url=None,
                  urllib3_proxy_kwargs=None,
                  connect_timeout=5.,
-                 read_timeout=5.):
+                 read_timeout=5.,
+                 retries=3):
```

Comment:
I wonder if we should have one or all of the arguments: `retries`, `backoff_factor` and/or `urllib3_req_kwargs`.
Personally I lean towards `retries` and `urllib3_req_kwargs` for the following advantages:
1. Simple things are visible and easily accessible.
1. It hints that everything else this are advanced options, but still controllable/accessible by the user.

So basically this will allow users understanding the nuances of networking and/or urllib3 use it. Others, will stay away.

However, I'd like to hear the other maintainers take on this.

Reply:
retries and urllib3_retry_kwargs sounds good to me for those reasons :)